cmake/match.py

```diff
@@ -35,9 +43,26 @@
         else:
             pattern += part
 
-    input_line = input_lines[i].strip()
+    # empty input file or end of input file, from now on match file must be optional
+    if not input_lines:
+        if optional_line is True:
+            continue
+        else:
+            print("End of input file or empty file.")
+            print("expected: " + match_line.strip())
+            sys.exit(1)
+
+    input_line = input_lines[input_idx].strip()
     if not re.fullmatch(pattern, input_line):
-        print(f"Line {i+1} does not match".format(i+1))
-        print("is:       " + input_line)
-        print("expected: " + match_line.strip())
-        sys.exit(1)
+        if optional_line is True:
```

Comment:
As I understand, this makes optional lines hard to check for - they can contain literally anything and the test can pass. Can we make opt lines required given a certain test scenario or setup?

Reply:
That would require us to somehow make the lines "conditional", so something like `{{skip if compiler==clang}}`. While that might be useful, it's imho out of scope for this work. And no, optional lines can't be anything - this `continue` here will only move forward the match file, not the input file, so if the line doesn't match either the optional line or the next one, it will fail.